src/static/js/modules/post-filters-form-validation.js

```diff
@@ -25,31 +25,41 @@ function init() {
     // Check that the date isn't empty,
     // and set it to the other set date if it is.
     // If it's still empty, return an empty string.
-    // If year or month is missing, the current month or year are added.
+    // If day, month, or year is missing, the current day,
+    // month, or year are added.
     // @param targetDate {String} Date string to format.
     // @param compData {String} Other date in the date range we're checking.
     // @return {String} Formatted targetDate or an empty string.
-    //   Throws an error if the expected YYYY-MM format is wrong.
+    // Throws an error if the expected MM/DD/YYYY format is wrong.
     function formatDateString( targetDate, compDate ) {
       var formattedDate = targetDate === '' ? compDate : targetDate;
       if ( formattedDate === '' ) return formattedDate;
 
-      if ( /^\d{4}$/.test( formattedDate ) ) {
-        var month = String( new Date().getMonth() + 1 );
-        if ( month.length !== 2 ) month = '0' + month;
-        formattedDate += '-' + month;
-      } else if ( /^\d{2}$/.test( formattedDate ) ) {
-        var year = new Date().getFullYear();
-        formattedDate = year + '-' + formattedDate;
-      } else if ( /^\d{4}-\d{2}$/.test( formattedDate ) === false ) {
-        var msg = 'Unexpected date format! Should be in YYYY-MM format.';
-        throw new Error( msg );
+      var day = '01';
+      var month = String( new Date().getMonth() + 1 );
+      var year = new Date().getFullYear();
+      var dateArray = ( /(?:(\d{1,2})\/)?(\d{1,2})\/(\d{2,4})/ )
+                      .exec( formattedDate );
+      var isArray = Array.isArray || function( obj ) {
+        return toString.call( obj ) === '[object Array]';
+      };
+
+      if ( isArray( dateArray ) ) {
+        day = dateArray[2] || day;
+        month = dateArray[1] || day;
+        year = dateArray[3] || year;
+      } else {
+
+        // TODO: call the notifier when invalid criteria passed.
+        var msg = 'Unexpected date format! Should be in MM/DD/YYYY format.';
+        throw ( msg );
```

Comment:
You may be able to use the validate.js plugin. We could definitely add more testing "types" to the cf_formValidation if needed.